classes/property_list.php

```diff
@@ -0,0 +1,295 @@
+     * @throws \CFPropertyList\IOException
+     * @throws \CFPropertyList\PListException
+     * @throws \DOMException
+     */
+    public function __construct(string $xml) {
+        $this->cfpropertylist = new CFPropertyList();
+        $this->cfpropertylist->parse($xml, CFPropertyList::FORMAT_XML);
+    }
+
+    /**
+     * Create an empty property list.
+     *
+     * @return property_list Return a property list with a empty root dictionary set up.
+     *
+     * @throws \CFPropertyList\IOException
+     * @throws \CFPropertyList\PListException
+     * @throws \DOMException
+     */
+    public static function create() {
+        $cfpropertylist = new CFPropertyList();
+        // Add main dict.
+        $cfpropertylist->add(new CFDictionary([]));
+        return new self($cfpropertylist->toXML());
+    }
+
+    /**
+     * Add a new element to the root dictionary element.
+     *
+     * @param string $key Key to assign to new element.
+     * @param CFType $element The new element. May be a collection such as an array.
+     */
+    public function add_element_to_root(string $key, CFType $element) {
+        // Get the PList's root dictionary and add new element.
+        $this->cfpropertylist->getValue()->add($key, $element);
+    }
+
+    /**
+     * Get value of element identified by key.
+     *
+     * @param string $key Key of element.
+     * @return mixed Value of element found, or null if none found.
+     */
+    public function get_element_value(string $key) {
+        $result = null;
+        $this->plist_map( function($elvalue, $elkey, $parent) use ($key, &$result) {
+            // Convert date to iso 8601 if date object.
+            if ($key === $elkey) {
+                $result = $elvalue->getValue();
+            }
+        }, $this->cfpropertylist->getValue());
+        // Turn PList dicts and arrays into PHP array for export.
+        if (is_array($result)) {
+            $result = new CFDictionary($result); // Convert back to CFDictionary so serialization is recursive.
+            $result = $result->toArray(); // Serialize.
+        }
+        return $result;
+    }
+
+    /**
+     * Update the value of any element with matching key.
+     *
+     * @param string $key Key of element to update.
+     * @param mixed $value  Value to update element with.
+     */
+    public function update_element_value(string $key, $value) {
+        $this->plist_map( function($elvalue, $elkey, $parent) use ($key, $value) {
+            // Set new value.
+            if ($key === $elkey) {
+                $parent->get($elkey)->setValue($value);
+            }
+        }, $this->cfpropertylist->getValue());
+    }
+
+    /**
+     * Update the array of any dict or array element with matching key.
+     *
+     * Will replace array.
+     *
+     * @param string $key Key of element to update.
+     * @param array $value  Array to update element with.
+     */
+    public function update_element_array(string $key, array $value) {
```

Comment:
Should it be public?

Reply:
If it is please add some negative tests. What would happen if value would be some random array?

Reply:
Added negative test. When trying to update, array is validated to make sure it can be used.